.github/workflows/npm-publish.yml

```diff
@@ -3,6 +3,8 @@ on:
   push:
     branches:
       - main
+    paths:
+    - 'packages/**'
```

Comment:
This will avoid publishing new revisions if there aren't any changes to actual packages.